lib/parser/unparse.cc

```diff
@@ -194,18 +194,18 @@ class UnparseVisitor {
       }
     }
     Put(QuoteCharacterLiteral(
-        std::get<std::string>(x.t), true, backslashEscapes_));
-  }
-  void Before(const HollerithLiteralConstant &x) {
-    std::optional<std::size_t> chars{CountCharacters(x.v.data(), x.v.size(),
-        encoding_ == Encoding::EUC_JP ? EUC_JPCharacterBytes
-                                      : UTF8CharacterBytes)};
-    if (chars.has_value()) {
-      Unparse(*chars);
-    } else {
-      Unparse(x.v.size());
-    }
+        std::get<std::string>(x.t), backslashEscapes_, Encoding::LATIN_1));
```

Comment:
Interpreting Hollerith requires decoding multi-byte sequences in the prescanner, since the explicit length of the Hollerith is in units of decoded characters, not bytes.  Quoted character literals are now run through decoding and re-encoding so that any escape sequences in them are normalized.